src/libprojectM/Audio/WaveformAligner.cpp

```diff
@@ -31,104 +31,101 @@ WaveformAligner::WaveformAligner()
-                 * i = 0.32*compareSamples
-                 *
-                 * The weight distribution is symmetric so the falling side gives:
-                 *
-                 * i = 0.68*compareSamples
-                 */
-                weightRef = (weightRef - 0.8f) * 5.0f + 0.8f;
-
-                // Clamp
-                // Needed because the TWEAK above results in weights from -3.2 to 1.8
-                if (weightRef > 1.0f)
-                {
-                    weightRef = 1.0f;
-                }
-                if (weightRef < 0.0f)
-                {
-                    weightRef = 0.0f;
-                }
+                weightRef = static_cast<float>(sample * 2) / static_cast<float>(compareSamples);
             }
-
-            uint32_t sample{};
-            // The code below also is only needed because of the TWEAK above, which zeroes
-            // a total of 64% of the weights.
-            while (m_aligmentWeights[octave][sample] == 0 && sample < compareSamples)
+            else
             {
-                sample++;
+                weightRef = static_cast<float>((compareSamples - 1 - sample) * 2) / static_cast<float>(compareSamples);
             }
-            m_firstNonzeroWeights[octave] = sample;
 
-            sample = compareSamples - 1;
-            while (m_aligmentWeights[octave][sample] == 0 && compareSamples > 1)
+            /*
+             * TWEAK how much the center matters, vs. the edges:
+             *
+             * weight[i] = 5.0*((2*i/compareSamples) - 0.8) + 0.8
+             * Solving for weight[i] == 0 we get:\
+             *
+             * 2*i/compareSamples = -0.8/5 + 0.8
+             * i = 0.32*compareSamples
+             *
+             * The weight distribution is symmetric so the falling side gives:
+             *
+             * i = 0.68*compareSamples
+             */
+            weightRef = (weightRef - 0.8f) * 5.0f + 0.8f;
+
+            // Clamp
+            // Needed because the TWEAK above results in weights from -3.2 to 1.8
+            if (weightRef > 1.0f)
             {
-                sample--;
+                weightRef = 1.0f;
             }
-            m_lastNonzeroWeights[octave] = sample;
+            if (weightRef < 0.0f)
+            {
+                weightRef = 0.0f;
+            }
+        }
+
+        uint32_t sample{};
+        // The code below also is only needed because of the TWEAK above, which zeroes
+        // a total of 64% of the weights.
+        while (m_aligmentWeights[octave][sample] == 0 && sample < compareSamples)
+        {
+            sample++;
         }
+        m_firstNonzeroWeights[octave] = sample;
+
+        sample = compareSamples - 1;
+        while (m_aligmentWeights[octave][sample] == 0 && compareSamples > 1)
+        {
+            sample--;
+        }
+        m_lastNonzeroWeights[octave] = sample;
     }
+}
 
+int WaveformAligner::CalculateOffset(std::vector<WaveformBuffer> &newWaveformMips)
```

Comment:
Better align reference/pointer designators to the type (or ideally use the .clang-format file in the root to format the code automatically).
`int WaveformAligner::CalculateOffset(std::vector<WaveformBuffer>& newWaveformMips)`